src/VecSim/algorithms/hnsw/hnswlib.h

```diff
@@ -806,27 +806,35 @@ bool HierarchicalNSW<dist_t>::removePoint(const labeltype label) {
     // replace the entry point with another one, if we are deleting the current entry point.
     if (element_internal_id == enterpoint_node_) {
         assert(element_top_level == maxlevel_);
-        linklistsizeint *top_level_list = get_linklist_at_level(element_internal_id, maxlevel_);
-        unsigned short list_len = getListCount(top_level_list);
-        while (list_len == 0) {
-            maxlevel_--;
-            if (maxlevel_ < 0) {
+        // Sets the (arbitrary) new entry point.
+        while (element_internal_id == enterpoint_node_) {
+            linklistsizeint *top_level_list = get_linklist_at_level(element_internal_id, maxlevel_);
+
+            if (getListCount(top_level_list)) {
```

Comment:
please be verbose here
```suggestion
            if (getListCount(top_level_list) > 0) {
```